lib/pry/pry_class.rb

```diff
@@ -1,7 +1,17 @@
 require 'pry/config'
 class Pry
 
-  HOME_RC_FILE = ENV["PRYRC"] || "~/.pryrc"
+  HOME_RC_FILE = proc do
+    if ENV.has_key?('PRYRC')
+      ENV['PRYRC']
+    elsif File.exist?(File.expand_path('~/.pryrc'))
+      '~/.pryrc'
+    elsif ENV.has_key?('XDG_CONFIG_HOME') && ENV['XDG_CONFIG_HOME'] != ''
+      ENV['XDG_CONFIG_HOME'] + '/pry/config'
+    else
+      '~/.config/pry/config'
```

Comment:
Like some comments? Sure, how about

```ruby
elsif ENV.has_key?('XDG_CONFIG_HOME') && ENV['XDG_CONFIG_HOME'] != ''
  # See XDG Base Directory Specification at
  # https://standards.freedesktop.org/basedir-spec/basedir-spec-0.8.html
  ENV['XDG_CONFIG_HOME'] + '/pry/config'
else
```

?

As for the path, the specification doesn't seem to specify anywhere whether it should be `$XDG_CONFIG_HOME/foorc` or `$XDG_CONFIG_HOME/foo/config`; I think that's just what everybody is doing when following the specification. For example, [this is what Git has been doing](https://git-scm.com/docs/git-config/2.13.0#FILES), and some application on your system may also be doing that. I agree with it, since this looks nicer if an application has more than one configuration file (for example, Git has two: `config` and `ignore` file).

Fun fact: the specification I quote should be version 0.8, last updated 2014-04-04 11:02, but the document itself says

> Version 0.7
>
> 24th November 2010

Aha! I demand a pull request to them!

![Fix everything!](https://cdn.meme.am/instances/54604398.jpg)